Cabal/Distribution/PackageDescription/Parsec/FieldDescr.hs

```diff
@@ -0,0 +1,580 @@
+             homepage               (\val pkg -> pkg{homepage=val})
+   , simpleField "package-url"
+             showFreeText           freeTextFieldParser
+             pkgUrl                 (\val pkg -> pkg{pkgUrl=val})
+   , simpleField "bug-reports"
+             showFreeText           freeTextFieldParser
+             bugReports             (\val pkg -> pkg{bugReports=val})
+   , simpleField "synopsis"
+             showFreeText           freeTextFieldParser
+             synopsis               (\val pkg -> pkg{synopsis=val})
+   , simpleField "description"
+             showFreeText           freeTextFieldParser
+             description            (\val pkg -> pkg{description=val})
+   , simpleField "category"
+             showFreeText           freeTextFieldParser
+             category               (\val pkg -> pkg{category=val})
+   , simpleField "author"
+             showFreeText           freeTextFieldParser
+             author                 (\val pkg -> pkg{author=val})
+   , listField "tested-with"
+             showTestedWith         parsecTestedWith
+             testedWith             (\val pkg -> pkg{testedWith=val})
+   , listFieldWithSep vcat "data-files"
+             showFilePath           parsecFilePath
+             dataFiles              (\val pkg -> pkg{dataFiles=val})
+   , simpleField "data-dir"
+             showFilePath           parsecFilePath
+             dataDir                (\val pkg -> pkg{dataDir=val})
+   , listFieldWithSep vcat "extra-source-files"
+             showFilePath           parsecFilePath
+             extraSrcFiles          (\val pkg -> pkg{extraSrcFiles=val})
+   , listFieldWithSep vcat "extra-tmp-files"
+             showFilePath           parsecFilePath
+             extraTmpFiles          (\val pkg -> pkg{extraTmpFiles=val})
+   , listFieldWithSep vcat "extra-doc-files"
+             showFilePath           parsecFilePath
+             extraDocFiles          (\val pkg -> pkg{extraDocFiles=val})
+   ]
+
+-- | Store any fields beginning with "x-" in the customFields field of
+--   a PackageDescription.  All other fields will generate a warning.
+storeXFieldsPD :: UnknownFieldParser PackageDescription
+storeXFieldsPD f val pkg | beginsWithX f =
+    Just pkg { customFieldsPD = customFieldsPD pkg ++ [(fromUTF8BS f, trim val)] }
+storeXFieldsPD _ _ _ = Nothing
+
+-------------------------------------------------------------------------------
+-- Library
+-------------------------------------------------------------------------------
+
+libFieldDescrs :: [FieldDescr Library]
+libFieldDescrs =
+    [ listFieldWithSep vcat "exposed-modules" disp (parsecMaybeQuoted parsec)
+        exposedModules (\mods lib -> lib{exposedModules=mods})
+    , commaListFieldWithSep vcat "reexported-modules" disp parsec
+        reexportedModules (\mods lib -> lib{reexportedModules=mods})
+
+{-
+  , listFieldWithSep vcat "required-signatures" disp parseModuleNameQ
```

Comment:
We do run them (this is how I caught `autogen-modules` e.g.). Or are backpack package tests in `cabal-install`?